### ISSUE TITLE
Flaky hanging test: Running_AppHost_with_GUI_Reports_Errors_In_Window

### ISSUE THREAD

**Issue by dagood** [dotnet/core-setup#6345]
Repros of https://github.com/dotnet/core-setup/issues/6117:

https://dev.azure.com/dnceng/internal/_build/results?buildId=180369

```
  [xUnit.net 00:07:38.5655064]     Microsoft.DotNet.CoreSetup.Test.HostActivation.StandaloneAppActivation.Running_AppHost_with_GUI_Reports_Errors_In_Window [FAIL]
  Failed   Microsoft.DotNet.CoreSetup.Test.HostActivation.StandaloneAppActivation.Running_AppHost_with_GUI_Reports_Errors_In_Window
  Error Message:
   Waited 90000 milliseconds for the popup window on process 2396, but none was found.
  
  Expected: True
  Actual:   False
  Stack Trace:
     at Microsoft.DotNet.CoreSetup.Test.HostActivation.StandaloneAppActivation.WaitForPopupFromProcess(Process process, Int32 timeout) in F:\workspace\_work\1\s\src\test\HostActivationTests\StandaloneAppActivation.cs:line 358
     at Microsoft.DotNet.CoreSetup.Test.HostActivation.StandaloneAppActivation.Running_AppHost_with_GUI_Reports_Errors_In_Window() in F:\workspace\_work\1\s\src\test\HostActivationTests\StandaloneAppActivation.cs:line 255
```

https://dev.azure.com/dnceng/internal/_build/results?buildId=179626

```
  [xUnit.net 00:02:43.7896633]     Microsoft.DotNet.CoreSetup.Test.HostActivation.StandaloneAppActivation.Running_AppHost_with_GUI_Reports_Errors_In_Window_and_Traces [FAIL]
  Failed   Microsoft.DotNet.CoreSetup.Test.HostActivation.StandaloneAppActivation.Running_AppHost_with_GUI_Reports_Errors_In_Window_and_Traces
  Error Message:
   Waited 90000 milliseconds for the popup window on process 11244, but none was found.
  
  Expected: True
  Actual:   False
  Stack Trace:
     at Microsoft.DotNet.CoreSetup.Test.HostActivation.StandaloneAppActivation.WaitForPopupFromProcess(Process process, Int32 timeout) in F:\workspace\_work\1\s\src\test\HostActivationTests\StandaloneAppActivation.cs:line 358
     at Microsoft.DotNet.CoreSetup.Test.HostActivation.StandaloneAppActivation.Running_AppHost_with_GUI_Reports_Errors_In_Window_and_Traces() in F:\workspace\_work\1\s\src\test\HostActivationTests\StandaloneAppActivation.cs:line 281
```

This seems to be a hang. (https://github.com/dotnet/core-setup/pull/6168 set 30s as the new timeout, with 90s being a "presumed hang" longer timeout.)

@vitek-karas

**Comment by jeffschwMSFT** [comment]
cc @swaroop-sridhar

**Comment by vitek-karas** [comment]
This should be hopefully fixed by #6823 ... please reopen if this still happens.
